python_bindings/src/PyFunc.cpp

```diff
@@ -36,6 +36,25 @@ void define_set(py::class_<Func> &func_class) {
     ;
 }
 
+// See the usages below this function to see why we are specializing this function.
+template<typename RHS>
+void define_set_func_ref(py::class_<Func> &func_class) {
+    func_class
+        .def("__setitem__", [](Func &func, const FuncRef &lhs, const RHS &rhs) -> Stage {
+            return func(lhs) = rhs;
+        });
+}
+
+// Special case: there is no implicit conversion of double in C++ Halide
+// but there is implicit conversion of double in Python.
+template<>
+void define_set_func_ref<double>(py::class_<Func> &func_class) {
+    func_class
+        .def("__setitem__", [](Func &func, const FuncRef &lhs, const double &rhs) -> Stage {
+            return func(lhs) = Expr(rhs);
```

Comment:
Should perhaps throw an error if rhs is not exactly representable as a float (we shouldn't implicitly lose precision). Not sure what the best way to check that is in a fast-math world.

Reply:
See #4247 for context

Reply:
Added some warnings. It seems that the precise content of warnings depend on #4247. @steven-johnson do you need help on that PR?

Reply:
I wasn't working on the PR (I had assumed that @mugwort-rc would finish it with the relevant comments, since they authored the original), but nothing much has happened. I'd be happy for you to resurrect it and apply a similar fix there :-)